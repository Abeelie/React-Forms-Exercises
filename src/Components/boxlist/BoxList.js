import React, { useState } from "react";
import Box from "../box/Box";
import BoxForm from "../boxform/BoxForm";
import { v4 as uuid } from "uuid";
import "./boxlist.css";

const BoxList = () => {
    const [boxList, setBoxList] = useState([]);

    const addBox = (initial) => {
        const newBox = { ...initial, id: uuid() };
        setBoxList(boxList => [...boxList, newBox]);
    }

    const removeBox = (id) => {
        setBoxList(boxList => boxList.filter(box => box.id !== id));
    }

    const showBoxs = boxList.map(box => (
        <Box
          key={uuid()}
          id={box.id}
          width={box.width}
          height={box.height}
          removeBox={removeBox}
          bgColor={box.bgColor}
        />
      ));

    return (
        <div className="BoxList">
          <BoxForm addBox={addBox} />
          {showBoxs}
        </div>
    );
}


export default BoxList