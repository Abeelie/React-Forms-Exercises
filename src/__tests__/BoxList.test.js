import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "../Components/boxlist/BoxList";


it("renders without crashing", function() {
    render(<BoxList />);
});
  
  it("matches snapshot", function() {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});
  
const addBox = (boxList, height = "100", width = "100", color = "blue") => {
    const bgColor = boxList.getByLabelText("Background-Color:");
    const inHeight = boxList.getByLabelText("Height:");
    const inWidth = boxList.getByLabelText("Width:");

    fireEvent.change(bgColor, { target: { value: color } });
    fireEvent.change(inHeight, { target: { value: height } });
    fireEvent.change(inWidth, { target: { value: width } });
   
    const button = boxList.getByText("Add box");
    fireEvent.click(button);
  }



it("add box", function() {
    const boxList = render(<BoxList />);

    addBox(boxList);
    const removeBtn = boxList.getByText("x");
    expect(removeBtn).toBeInTheDocument();

    expect(removeBtn.previousSibling).toHaveStyle(`
        background-color: blue;
        width: 100px;
        height: 100px;
    `);
});
  

it("remove a box", function() {
    const boxList = render(<BoxList />);
    addBox(boxList);
    
    const removeButton = boxList.getByText("x");
    fireEvent.click(removeButton);
    expect(removeButton).not.toBeInTheDocument();
});