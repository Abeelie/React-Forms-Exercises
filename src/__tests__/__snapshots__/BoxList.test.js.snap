// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h1>
      Box List
    </h1>
    <form
      class="form"
    >
      <label
        for="bgColor"
      >
        Background-Color: 
      </label>
      <input
        class="input"
        id="bgColor"
        name="bgColor"
        value=""
      />
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        class="input"
        id="width"
        name="width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        class="input"
        id="height"
        name="height"
        type="number"
        value=""
      />
      <button
        class="button"
      >
        Add box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
